odoo/tools/image.py

```diff
@@ -378,16 +378,13 @@ def image_fix_orientation(image):
         or the source image if no operation was applied
     :rtype: PIL.Image
     """
-    if (image.format or '').upper() == 'JPEG' and hasattr(image, '_getexif'):
```

Comment:
I obviously did it this way originally for a reason, especially the "only JPEG" part, but I can't remember for sure.

I think it has to do with earlier version of Pillow that we supported back then (v4-v5 probably) that didn't have the public `getexif` method (which I think is only available from v6). Those earlier version probably had this private method `_getexif` but only on JPEG hence the check.

The current PR would very likely break EXIF support for Pillow earlier than v6. Is that something we are ok with? Or should we keep the fallback to private method as well?

Reply:
> Or should we keep the fallback to private method as well?

I thing that would make sense e.g.

```python
getexif = getattr(image, 'getexif', None) or getattr(image, '_getexif', None)
if getexif:
    exif = getexif()
    ...
```

Reply:
It makes sense, I think.

Reply:
Could even fallback on an "empty" function to remove one layer of conditionals:
```python
exit = (getattr(image, 'getexif', None) or getattr(image, '_getexif', None) or lambda: None)()
if exif:
    ...
```
Though the clarity of tacking a pair of parens at the end is somewhat debatable.

Reply:
Ok for the fallback.

I was still just a bit concern with removing the JPEG requirement for the private `_getexif` method, but I double-checked Pillow code before v6 and it looks like only JPEG has the private method, so hopefully the condition was just there to avoid unnecessary call to `getattr`. 

Also note that `getexif` doesn't actually return the same value. `_getexif` is just a dict, but `getexif` returns a custom class `Exif(MutableMapping)`. That shouldn't be an issue for our purpose because we basically just call `get` which is available on the two, but something to keep in mind. In Pillow code where `getexif` exists, `_getexif` is made to return `dict(getexif())`.